src/test/java/ay2122s1_cs2103t_w16_2/btbb/logic/parser/AddressBookParserTest.java

```diff
@@ -123,6 +133,22 @@ public void parseCommand_findClient() throws Exception {
         assertEquals(new FindClientCommand(predicateCollection), command);
     }
 
+    @Test
+    public void parseCommand_findIngredient() throws Exception {
+        PredicateCollection<Ingredient> predicateCollection = new PredicateCollection<>();
+        predicateCollection.addPredicate(INGREDIENT_NAME_AVOCADO_BUTTER_PREDICATE);
+        predicateCollection.addPredicate(INGREDIENT_QUANTITY_5_550_PREDICATE);
+        predicateCollection.addPredicate(INGREDIENT_QUANTITY_5_TO_600_PREDICATE);
+        predicateCollection.addPredicate(INGREDIENT_UNIT_WHOLE_GRAMS_PREDICATE);
+        FindIngredientCommand command = (FindIngredientCommand) parser.parseCommand(FindIngredientCommand.COMMAND_WORD
+                + " "
+                + PREFIX_INGREDIENT_NAME + "Avocado Butter "
+                + PREFIX_INGREDIENT_QUANTITY + "5 550 "
+                + PREFIX_INGREDIENT_QUANTITY_FROM + "5 " + PREFIX_INGREDIENT_QUANTITY_TO + "600 "
+                + PREFIX_INGREDIENT_UNIT + "whole grams");
+        assertEquals(new FindIngredientCommand(predicateCollection), command);
+    }
```

Comment:
For the AddressBookParserTest, we also previously added the parseCommand_findClientNoKeywords_exceptionThrown and parseCommand_invalidFindOrderCommand_exceptionThrown methods to test wrong inputs. Should we have one for findIngredient also or should we remove the other two methods since we handle those error cases in FindClientCommandParserTest and FindOrderCommandParserTest.